File: go/spec/echo_service.go
package spec

type EchoService struct{}

func (service *EchoService) EchoBody(body *Message) (*EchoBodyResponse, error) {
	return &EchoBodyResponse{Ok: body}, nil
}

func (service *EchoService) EchoQuery(intQuery int, stringQuery string) (*EchoQueryResponse, error) {
	return &EchoQueryResponse{Ok: &Message{IntField: intQuery, StringField: stringQuery}}, nil
}

func (service *EchoService) EchoHeader(intHeader int, stringHeader string) (*EchoHeaderResponse, error) {
	return &EchoHeaderResponse{Ok: &Message{IntField: intHeader, StringField: stringHeader}}, nil
}

func (service *EchoService) EchoUrlParams(intUrl int, stringUrl string) (*EchoUrlParamsResponse, error) {
	return &EchoUrlParamsResponse{Ok: &Message{IntField: intUrl, StringField: stringUrl}}, nil
}

File: java-spring/src/main/java/test_service/implementations/EchoService.java
package test_service.implementations;

import org.springframework.stereotype.Service;
import test_service.models.Message;
import test_service.responses.*;
import test_service.services.IEchoService;

@Service
public class EchoService implements IEchoService {

	@Override
	public EchoBodyResponse echoBody(Message body) {
		return new EchoBodyResponse(body);
	}

	@Override
	public EchoQueryResponse echoQuery(int intQuery, String stringQuery) {
		return new EchoQueryResponse(new Message(intQuery, stringQuery));
	}

	@Override
	public EchoHeaderResponse echoHeader(int intHeader, String stringHeader) {
		return new EchoHeaderResponse(new Message(intHeader, stringHeader));
	}

	@Override
	public EchoUrlParamsResponse echoUrlParams(int intUrl, String stringUrl) {
		return new EchoUrlParamsResponse(new Message(intUrl, stringUrl));
	}
}
